snapshot/snapshot.sln

```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```

Comment:
Overall LGTM,  rather than adding individual solutions, please add a solution at root with all projects in them, it would be easier to manage.

Reply:
I agree.

Reply:
+1, So we have the common model class which is called `SocialMediaMessage`. I would create classlib project to hold this common models instead of defining multiple times in the multiple apps.